### PR TITLE
Add release notes for version 2.0.1

### PR DESCRIPTION
**Pull request by kotwanikunal**
Signed-off-by: Kunal Kotwani <kkotwani@amazon.com>

### Description
- Add release notes for 2.0.1
 
### Issues Resolved
- Part of [2165](https://github.com/opensearch-project/opensearch-build/issues/2165)
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch.release-notes-2.0.1.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## 2022-06-15 Version 2.0.1 Release Notes
+
+### Bug Fixes
+* [Type removal] Ignore _type field in bulk request ([#3504](https://github.com/opensearch-project/OpenSearch/pull/3504))
+* [Type removal] Revert removal of typed end-points for bulk, search, index APIs ([#3528](https://github.com/opensearch-project/OpenSearch/pull/3528))
+* [Type removal] Revert beat _type changes ([#3559](https://github.com/opensearch-project/OpenSearch/pull/3559))
```
Comment (dreamer-89) [inline release-notes/opensearch.release-notes-2.0.1.md:6]: These three commits can be removed. First two commit added typed end-points (and type support in bulk request), the third commit reverts the first two.

Comment (kotwanikunal) [reply release-notes/opensearch.release-notes-2.0.1.md]: Updated the PR

**Review by dreamer-89 — APPROVED** [review]
LGTM!

**Review by reta — APPROVED** [review]

**Comment by kotwanikunal** [comment]
```
2> REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testHighWatermarkNotExceeded" -Dtests.seed=F9A946563670180E -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=pt-BR -Dtests.timezone=Europe/Istanbul -Druntime.java=17
  2> java.lang.AssertionError: Mismatching shard routings: []
    Expected: a collection with size <1>
         but: collection size was <0>
        at __randomizedtesting.SeedInfo.seed([F9A946563670180E:1088A7E4B6B6D1E0]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:964)
        at org.opensearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.lambda$assertBusyWithDiskUsageRefresh$5(DiskThresholdDeciderIT.java:362)
        at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1049)
        at org.opensearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.assertBusyWithDiskUsageRefresh(DiskThresholdDeciderIT.java:355)
        at org.opensearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testHighWatermarkNotExceeded(DiskThresholdDeciderIT.java:188)
  2> NOTE: leaving temporary files on disk at: /var/CITOOL/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/search/server/build/testrun/internalClusterTest/temp/org.opensearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT_F9A946563670180E-004
  2> NOTE: test params are: codec=Lucene92, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=pt-BR, timezone=Europe/Istanbul
  2> NOTE: Linux 5.13.0-1014-aws amd64/Eclipse Adoptium 17.0.3 (64-bit)/cpus=72,threads=1,free=215626176,total=536870912
  2> NOTE: All tests run in this JVM: [TransportClusterStateActionDisruptionIT, IndexingClusterManagerFailoverIT, DiskThresholdDeciderIT]
```

Same issue as https://github.com/opensearch-project/OpenSearch/pull/3568#issuecomment-1156808924

**Review by tlfeng — APPROVED** [review]

**Pull request merged**